tests/e2e/hci_mesh_rwx_singlevc_topology.go

```diff
@@ -311,9 +327,368 @@ var _ = ginkgo.Describe("[hci-mesh-rwx-topology] Hci-Mesh-Topology-SingleVc", fu
 		}()
 
 		ginkgo.By("Verify volume placement, should match with the allowed topology specified")
-		_, _, datastoreUrlsRack3, _, err := fetchDatastoreListMap(ctx, client)
+		_, _, datastoreUrlsRack3, _, _, err := fetchDatastoreListMap(ctx, client)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for i := 0; i < len(pvs); i++ {
+			isCorrectPlacement := e2eVSphere.verifyPreferredDatastoreMatch(pvs[i].Spec.CSI.VolumeHandle, datastoreUrlsRack3)
+			gomega.Expect(isCorrectPlacement).To(gomega.BeTrue(), fmt.Sprintf("Volume provisioning has happened on the wrong "+
+				"datastore. Expected 'true', got '%v'", isCorrectPlacement))
+		}
+	})
+
+	/*
+		TESTCASE-3
+		Workload creation with higher level topology
+		Immediate, region-1 > zone-1 > building-1, storage policy specific to local datastore in rack-1
+
+		Steps:
+		1. Create Storage class with bindingmode set to Immediate and allowed topologies set
+		to region-1 > zone-1 > building-1. Implement a Storage policy specific to local datastore in rack-1
+		2. Create 5 PVCs with RWX access mode.
+		3. Create 5 deployments with 1 replicas using PVC created in step #2.
+		4. Wait for the PVCs to be created and reach the Bound state, and for the Pods to be
+		created and reach a running state.
+		5. Verify CSI honors storage policy mentioned in storageclass and provisions volume in rack1.
+		6. Volume provisioning happens only on local datastore i.e. vSAN Datastore of
+		Cluster-1 (rack-1) where vSAN FS is enabled on cluster1
+		7. Validate CNS metadata for each PVC and Pod.
+		8. Scale up all deployment pods to different replica set.
+		9. Perform cleanup by deleting pod,pvc and sc
+	*/
+
+	ginkgo.It("Deployment Pods attached to single RWX PVC with vSAN storage policy of rack-1 "+
+		"defined in SC along with top level allowed topology "+
+		"specified", ginkgo.Label(p0, file, vanilla, level5, level2, newTest), func() {
```

Comment:
topology label is required here?

Reply:
we don't have topology label, level2, level5 itself means that we are referring to topology different setups